gammapy/background/models.py

```diff
@@ -71,3 +77,491 @@ def evaluate(self, x, y):
+        return Quantity(energy_bin_centers, 'TeV')
+
+
+    def find_det_bin(self, det):
+        """Find the bin that contains the specified det (X, Y) pair.
+
+        Parameters
+        ----------
+        det : `~astropy.coordinates.Angle`
+        	det (X, Y) pair to search for
+
+        Returns
+        -------
+        bin_pos : `~int`
+		index of the det bin containing the specified det (X, Y) pair
+        bin_edges : `~astropy.units.Quantity`
+		det bin edges (x_lo, x_hi, y_lo, y_hi)
+        """
+        # check shape of det: only 1 pair is accepted
+        nvalues = len(det.flatten())
+        if nvalues != 2:
+            print("Expected exactly 2 values for det (X, Y), got {}.".format(nvalues))
+            raise IndexError
+
+        # check that the specified det is within the boundaries of the model
+        det_extent = self.image_extent
+        if not ((det_extent[0] <= det[0]) and (det[0] < det_extent[1])) or not ((det_extent[2] <= det[1]) and (det[1] < det_extent[3])):
+            print("Specified det {0} is outside the boundaries {1}.".format(det, det_extent))
+            raise ValueError
+
+        det_edges_low = self.det_bins[:-1]
+        det_edges_high = self.det_bins[1:]
+        detx_mask = (det_edges_low <= det[0]) & (det[0] < det_edges_high)
+        dety_mask = (det_edges_low <= det[1]) & (det[1] < det_edges_high)
+        bin_ids = np.arange(len(self.det_bins)-1)
+        bin_pos_x = bin_ids[detx_mask]
+        bin_pos_y = bin_ids[dety_mask]
+        bin_pos = np.array([bin_pos_x[0], bin_pos_y[0]])
+        bin_detx_low = det_edges_low[detx_mask]
+        bin_dety_low = det_edges_low[dety_mask]
+        bin_detx_high = det_edges_high[detx_mask]
+        bin_dety_high = det_edges_high[dety_mask]
+        bin_edges = Angle([bin_detx_low, bin_detx_high, bin_dety_low, bin_dety_high])
+
+        return bin_pos, bin_edges.flatten()
+
+
+    def find_energy_bin(self, energy):
+        """Find the bin that contains the specified energy value.
+
+        Parameters
+        ----------
+        energy : `~astropy.units.Quantity`
+        	energy to search for
+
+        Returns
+        -------
+        bin_pos : `~int`
+		index of the energy bin containing the specified energy
+        bin_edges : `~astropy.units.Quantity`
+		energy bin edges [E_min, E_max)
+        """
+        # check shape of energy: only 1 value is accepted
+        nvalues = len(energy.flatten())
+        if nvalues != 1:
+            print("Expected exactly 1 value for energy, got {}.".format(nvalues))
+            raise IndexError
+
+        # check that the specified energy is within the boundaries of the model
+        energy_extent = self.spectrum_extent
+        if not (energy_extent[0] <= energy) and (energy < energy_extent[1]):
+            print("Specified energy {0} is outside the boundaries {1}.".format(energy, energy_extent))
+            raise ValueError
+
+        energy_edges_low = self.energy_bins[:-1]
```

Comment:
Again, it should be one or two lines of code to find the index of the right energy bin.
Examples: https://github.com/gammapy/gammapy/search?utf8=%E2%9C%93&q=argmin

Note that @kingj90 has started working on a nice generic solution to energy binning handling here:
https://github.com/gammapy/gammapy/pull/290
So don't put too much work into this now ... it'll probably have to be adapted to use what @kingj90 implemented in the coming weeks.